QWeb/internal/secrets.py

```diff
@@ -47,23 +67,24 @@ def _filtered_start_keyword(keyword):
     # pylint: disable=protected-access, global-statement
     global log_level
     apply_filter = keyword.kwname in filtered_keywords
+    original_args = keyword.args
     if apply_filter:
         par_index, secret = filtered_keywords[keyword.kwname]
-        keyword = keyword.deepcopy()
-        keyword.args = list(keyword.args)
+        censored_args = list(keyword.args)
         if secret == 'hint':
-            keyword.args[par_index] = 'SECRET'
+            censored_args[par_index] = 'SECRET'
         else:
-            keyword.args[par_index] = "*" * 5
+            censored_args[par_index] = "*" * 5
+
+        _replace_keyword_args(keyword, tuple(censored_args))
```

Comment:
This block of code is duplicated in `_filtered_end_keyword()`

Maybe turn it into a function?
`_hide_keyword_arg_values(keyword)`

Reply:
ok, will update soon.

Reply:
resolved by new commit